include/zephyr/rtio/rtio.h

```diff
@@ -313,8 +389,44 @@ struct rtio {
 
 	/* Completion queue */
 	struct rtio_cq *cq;
+
+#ifdef CONFIG_RTIO_SYS_MEM_BLOCKS
+	/* Memory pool associated with this RTIO context. */
+	struct sys_mem_blocks *mempool;
+	/* The size (in bytes) of a single block in the mempool */
+	uint32_t mempool_blk_size;
+	/* Map of allocated starting blocks */
+	struct rtio_mempool_map_entry *mempool_map;
+#endif /* CONFIG_RTIO_SYS_MEM_BLOCKS */
 };
 
+/** The memory partition associated with all RTIO context information */
+extern struct k_mem_partition rtio_partition;
+
+/**
+ * @brief Compute the mempool block index for a given pointer
+ *
+ * @param[in] r RTIO contex
+ * @param[in] ptr Memory pointer in the mempool
+ * @return Index of the mempool block associated with the pointer. Or UINT16_MAX if invalid.
+ */
+static inline uint16_t __rtio_compute_mempool_block_index(const struct rtio *r, const void *ptr)
```

Comment:
This function seems to be only used when `RTIO_SYS_MEM_BLOCKS` is configured.  Do we want to move the `ifdef` around this function?  Maybe create a new file `rtio_mempool.h/c` for new mempool functionality?

Reply:
I moved the ifdef, but I think I want to keep it all in one place, it gets weird with the __syscall